src/quickgui/qgsquickmapcanvasmap.cpp

```diff
@@ -14,26 +14,26 @@
  ***************************************************************************/
 
 #include <QQuickWindow>
-#include <QScreen>
 #include <QSGSimpleTextureNode>
-#include <QtConcurrent>
-
-#include "qgslabelingresults.h"
-#include "qgsmaprendererparalleljob.h"
-#include "qgsmessagelog.h"
-#include "qgspallabeling.h"
-#include "qgsproject.h"
-#include "qgsvectorlayer.h"
-#include "qgis.h"
+#include <QScreen>
+#include <qgis.h>
+#include <qgsexpressioncontextutils.h>
+#include <qgsmaprenderercache.h>
+#include <qgsmaprendererparalleljob.h>
+#include <qgsmessagelog.h>
+#include <qgspallabeling.h>
+#include <qgsproject.h>
+#include <qgsvectorlayer.h>
+#include <qgslabelingresults.h>
 
 #include "qgsquickmapcanvasmap.h"
 #include "qgsquickmapsettings.h"
-#include "qgsexpressioncontextutils.h"
 
 
 QgsQuickMapCanvasMap::QgsQuickMapCanvasMap( QQuickItem *parent )
   : QQuickItem( parent )
-  , mMapSettings( new QgsQuickMapSettings() )
+  , mMapSettings( std::make_unique<QgsQuickMapSettings>() )
+  , mCache( std::make_unique<QgsMapRendererCache>() )
```

Comment:
AFICS that was always null before and disfunctional?

Reply:
@m-kuhn , correct, that was my finding when implementing cache a while back.